packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart

```diff
@@ -82,7 +82,10 @@ abstract class SharedPreferencesStorePlatform extends PlatformInterface {
   Future<Map<String, Object>> getAll();
 
   /// Returns all key/value pairs persisting in this store that have given [prefix].
-  Future<Map<String, Object>> getAllWithPrefix(String prefix) {
+  Future<Map<String, Object>> getAllWithPrefix(
+    String prefix, {
+    Set<String>? allowList,
```

Comment:
This is a breaking change; you need to make a new method instead (ideally with a parameter object in case we want to add other things later). For maximal backwards compatibility, you could the default only throw if an allowList is provided, and otherwise have it delegate to the existing `getAllWithPrefix`.

https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-platform-interface-method-parameters